src/context-selection/attribute-option-combo-selector-bar-item/use-attribute-params.js

```diff
@@ -0,0 +1,38 @@
+import { useMetadata } from '../../metadata/index.js'
+import {
+    getDataSetById,
+    getCategoryComboById,
+} from '../../metadata/selectors.js'
+import { useContextSelection } from '../use-context-selection/index.js'
+
+/**
+ * Finds the attributeComboId for the current attribute-option selection
+ */
+
+export const useAttributeParams = () => {
+    const { data } = useMetadata()
+    const [{ dataSetId, attributeOptionComboSelection }] = useContextSelection()
+
+    if (data && dataSetId) {
+        const dataSet = getDataSetById(data, dataSetId)
+        const catComboId = dataSet.categoryCombo.id
+        const categoryCombo = getCategoryComboById(data, catComboId)
+
+        const selectedOptions = Object.values(attributeOptionComboSelection)
+        // all categories in catCombo should have a selection
+        const validSelection =
+            categoryCombo.categories.length === selectedOptions.length
+
+        return {
+            attributeCombo: catComboId,
+            attributeOptions: selectedOptions,
+            validSelection,
+        }
```

Comment:
I think the comparison and `validSelection` return value belong to the `useIsValidSelection` hook, not into this one

Reply:
I'm not sure I agree. That would mean we have to duplicate this code in `useIsValidSelection`-hook. We already have access to the categoryCombo in this hook, so I choose to expose `validSelection`, and use that in the `useIsValidSelection`-hook instead. If not we have to introduce metadata and the selectors in the other hook.

Reply:
> That would mean we have to duplicate this code in useIsValidSelection-hook

You're talking about the code to get the `categoryCombo`, right? I don't think it's an issue. If you prefer to keep the code dry, you could extract that into its own hook or (cached?) selector.

The reason why I prefer changing this is because I think the conceptual model is more important that a bit of code duplication or creating an extra hook or selector for this. Conceptually `validSelection` is just somewhat related to "attribute params", but not derived from it, even though we have all the values necessary for the `validSelection` calculation. `useAttributeParams` does not indicate that it will return `validSelection`, plus it does two things at the same time, which is an odd and normally discouraged code pattern (despite being a very mild issue in this case)

> If not we have to introduce metadata and the selectors in the other hook.

I don't see the issue with that, we're talking about 5 or 6 lines of code, right? And the data itself comes from memoized selectors.

Reply:
> Conceptually validSelection is just somewhat related to "attribute params", but not derived from it 
It actually is this. The `validSelection` here refers to if the `attrubute-params` is valid, not the entire context-selection. 

I can change this though, as it was mostly because I implemented this way of checking if it's valid just for attribute-params before I implemented the `useIsValidSelection`-hook, so I just figured I could re-use that code...

>  plus it does two things at the same time, which is an odd and normally discouraged code pattern

I can see that it comes off that way now, since we have the `useIsValidSelection`-hook, but previously I used this together with getting the `attributeCombo` and `attributeOptions`, which made this a very ergonomic API. 

I don't think custom hooks necessarily should "just do one thing", or that this case even fulfills that - it's just a helper., or some metadata about the data you return, if you will. Especially when it returns an object either way, it's no problem imo to return a helper property like this, when you have all the orchestration to calculate that prop around already.

Reply:
> not the entire context-selection

Ah, that's my bad then! I thought it's vor the entire context as the other hook was called `useIsValidSelection`. I saw that you already changed the `useIsValidHook` to include the other parameters as well :muscle: 

> I don't think custom hooks necessarily should "just do one thing", or that this case even fulfills that - it's just a helper, or some metadata about the data you return, if you will.

So, considering that I thought `validSelection` refers to the entire selection being valid, I would've disagreed with this, but it doesn't matter for this PR anymore as I misunderstood the purpose of the value. IMO even helpers should never do more than the name suggests. Whether the attribute params are valid or not could be seen as meta info of the params themselves, so I'd say the hooks still does "one thing" in this case, so it's fine as long as the other hook also considers the other context values.

---

I'll resolve this conversation as there's now a clear distinction between the return value of `useIsValidSelection` and the `validSelection` prop returned by the `useAttributeParams` hook.

Reply:
> IMO even helpers should never do more than the name suggests. Whether the attribute params are valid or not could be seen as meta info of the params themselves, so I'd say the hooks still does "one thing" in this case, so it's fine as long as the other hook also considers the other context values. 

When I mentioned "helper", I referred to the `isValid`-property like you suggest here. Not the entire hook. See now that is not so clear :)